nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/StandardProcessSession.java

```diff
@@ -2308,18 +2308,46 @@ public void exportTo(final FlowFile source, final Path destination, final boolea
     public void exportTo(final FlowFile source, final OutputStream destination) {
         validateRecordState(source);
         final StandardRepositoryRecord record = records.get(source);
+
         try {
-            if (record.getCurrentClaim() == null) {
```

Comment:
Can you explain the reasoning behind removing this? If there is no content claim, i think it still makes sense to return immediately. This would happen, for instance, with some source processors such as ListFile that don't actually write any content to the FlowFile.